hydrogym/ts.py

```diff
@@ -284,6 +285,58 @@ def rmatvec(q_vec):
                 B[:, i] = get_array(Bi)
             return A, B
 
+class Torque_IPCS(IPCS):
+    def __init__(self, flow: FlowConfig, dt: float, debug=False, **kwargs):
+        super().__init__(flow, dt, debug=False, **kwargs)
+        self.control = 0.0 # init with no torque on system
+        self.omega = [0.0] # Cylinder starts at rest
+        self.k_damping = 1/self.flow.TAU
+        self.I_cm = 0.006172 # Moment of inertia about CoM of a Plexiglass Cylinder with a 2 inch radius and spanning a half meter test section of a wind tunnel
```

Comment:
> Awesome that you found an actual value for this! Has this been nondimensionalized w.r.t. mass and diameter of the cylinder?

So following up on this point, is there a reason we are normalizing by diameter and not radius? 

Just asking because the equation is `I_cm = M R**2 / 2`, so would it make more sense to normalize by R? and whether or not it's D or R, should we normalize by (R/D)**2 ?

Reply:
Either is fine in principle, but the rest of the flow configuration has been normalized with the length scale D (so that the cylinder has radius 0.5).  And I guess the mass scale should be set by the fluid density.  So I don't know what units the mass of this was in (if the radius was in inches, maybe lbs?), but start with the fluid density (I know this was a wind tunnel, but let's use water as the fluid here) in those units.  I'd convert everything to metric too, just to avoid wonky units.

Then rho=(mass)/(length)^3, and I=(mass)*(length)^2, so if `rho` is the density of water (in the correct units) and `D` is the diameter of the cylinder, I think you would just scale `I_cm` by `D**5/rho` to make it dimensionless... but check for yourself that that makes sense.

Reply:
Sure, I can do that. I do have some questions though.

The way I calculated I_cm was using some arbitrary numbers (density of plexiglass = 1.18 g/cm^3, length of cylinder = 1m). So by choosing these values, we are no longer simulating an arbitrary actuating cylinder, we are simulating a plexiglass cylinder with a length of 1 meter and a radius of 0.5. Since the radius of 0.5 m is built into the simulation, I don't see any problems there, but would it maybe be best to normalize the cylinder first by length and density of the plexiglass and then by the diameter? i.e. something like scaling `I_cm` by '1/(rho_plexiglass * L * D**4)' ? or are we okay with making the design choice that our cylinder will be of length 1m and "made" of plexiglass?

Also, I'm scaling by '1/(rho_plexiglass * L * D**4)' instead of the 'L*D**4/rho_plexiglass` format that you suggested since the fully expanded equation for I_cm would be `I_cm = (rho_plexiglass * L * pi * R**2) * R**2 / 2`

Reply:
Oh alright, I was confused by the "2 inch radius" in the comment.  Yeah, you're definitely right the third dimension should be length and not radius though.  I'll have to work this out on paper I think, but let's leave it for now.  We can always update the numeric value if need be.  At the end of the day as long as it seems like it gives pretty good results I think it's fine.

So just to be clear for future reference, the properties you're assuming are for a 3D cylinder with density 1.18 g/cm^3 spanwise length 1m, and radius 0.5m?  And then `I_cm = (rho_plexiglass * L * pi * R**2) * R**2 / 2`?

Reply:
Sorry about that, those comments are super confusing, and I'll clean them up in my next commit. I initially used other experimental papers as a point of reference, and they mostly used cylinders with a radius of ~2 inches so that's where that number comes from.

That equation is correct as well as the length and density. At the moment, I am using radius=0.5 according to the flow config, but I guess I can't say for sure that the radius is in units of meters. It's just given in the code as `self.rad = fd.Constant(0.5)` with no mention of units (or maybe R is intentionally left unitless?).

Reply:
Yeah it's meant to be dimensionless with respect to the diameter.  Okay I think this is good then - we can always tweak the numerical value if need be later on.